minio/signer.py

```diff
@@ -198,7 +198,7 @@ def sign_v4(method, url, region, headers=None, access_key=None,
 
     date = datetime.utcnow()
     headers['X-Amz-Date'] = date.strftime("%Y%m%dT%H%M%SZ")
-    headers['X-Amz-Content-Sha256'] = content_sha256
+    headers['X-Amz-Content-Sha256'] = content_sha256.decode('ascii')
```

Comment:
It seems pretty clumsy to keep `content_sha256` as `bytes` in one case and as `str` in the other. I am making a slightly better change to handle this case.